quantum/quantum.c

```diff
@@ -114,6 +114,39 @@ void tap_code16(uint16_t code) {
     unregister_code16(code);
 }
 
+void register_magic_code(uint8_t code) {
+    register_code(keycode_config(code));
+}
+
+void register_magic_code16(uint16_t keycode) {
+    if (keycode >= QK_MODS && keycode <= QK_MODS_MAX) {
+        keycode = (mod_config((keycode >> 8) & 0x1F) << 8|keycode_config(keycode & 0xFF));
+    }
+    register_code16(keycode_config(keycode));
```

Comment:
If `keycode` initially had some modifier bits set, this call to `keycode_config()` will attempt to modify it a second time — which may actually succeed if that `mod_config()` above removed all modifier bits (which could happen if the `keymap_config.no_gui` option is enabled). Either that `keycode_config()` should be moved to the else branch of the above condition, or the case of having bare keycode with no modifiers should just be called in the same way as having some modifiers (remove `keycode >= QK_MODS` above, and remove the second call to `keycode_config()`).

Reply:
The `register_code16(keycode_config(keycode));` won't modify it, if it has mod bits set.  It's looking for straight keycodes, which will not be validated by the function, so it should be safe to call.

https://github.com/drashna/qmk_firmware/blob/411fbe606e8a3e28ef54c671baec1227449700c5/quantum/keycode_config.c#L26-L114

It's not doing any processing, so ...  it should be fine to leave as is. 
(unless I'm too tired to comprehend properly)

Reply:
I could imagine some really strange configuration:
* `keymap_config.no_gui` is enabled, and a key swapping option like `keymap_config.swap_grave_esc` is also enabled
* `keycode` is `G(KC_ESC)` (using `MOD_LGUI` or `MOD_RGUI`, and one of keycodes which are swapped)

In this case `mod_config(MOD_LGUI)` will return 0, therefore `keycode` will become just `KC_GRV`, which will be converted back to `KC_ESC` by the second `keycode_config()` call. It's very unlikely that someone would ever make this configuration, but technically it is possible, and the code which chains `keycode_config()` calls just looks wrong.

You could just do this (for basic keycodes this would just pass 0 to `mod_config()`):
```c
    if (keycode <= QK_MODS_MAX) {
        keycode = (mod_config((keycode >> 8) & 0x1F) << 8|keycode_config(keycode & 0xFF));
    }
    register_code16(keycode);
```
(adding `keycode >= QK_BASIC && ` might cause warnings due to this part being always true).